File: loggalitic/src/main/java/com/android/oleksandrpriadko/loggalitic/analytics/converter/Converter.java
package com.android.oleksandrpriadko.loggalitic.analytics.converter;

import android.support.annotation.NonNull;
import android.support.annotation.StringDef;

import com.android.oleksandrpriadko.loggalitic.analytics.AnalyticsEvent;


/**
 * Created by Oleksandr Priadko.
 * 7/9/17
 * <p>
 * Convert generic event to concrete implementation.
 */

public abstract class Converter<E> {

    public abstract E convert(@NonNull AnalyticsEvent analyticsEvent);

    public abstract Class getType();

}

File: loggalitic/src/main/java/com/android/oleksandrpriadko/loggalitic/analytics/converter/Converter.kt
package com.android.oleksandrpriadko.loggalitic.analytics.converter

import com.android.oleksandrpriadko.core.policy.AnalyticsEvent


/**
 * Convert generic event to concrete implementation.
 */

abstract class Converter<E> {

    abstract fun getType(): Class<*>

    abstract fun convert(analyticsEvent: AnalyticsEvent): E

}
